website/addons/dmptool/model.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+
+from modularodm import fields
+
+from framework.exceptions import HTTPError
+
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase, exceptions,
+)
+from website.addons.base import StorageAddonBase
+
+from website.addons.dmptool.serializer import DmptoolSerializer
+
+ROOT_FOLDER_NAME = '/'
+
+
+class DmptoolProvider(object):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'Dmptool'
+    short_name = 'dmptool'
+    serializer = DmptoolSerializer
+
+    def __init__(self, account=None):
+        super(DmptoolProvider, self).__init__()
+        # provide an unauthenticated session by default
+        self.account = account
+
+    def __repr__(self):
+        return '<{name}: {status}>'.format(
+            name=self.__class__.__name__,
+            status=self.account.provider_id if self.account else 'anonymous'
+        )
+
+class AddonDmptoolUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+class AddonDmptoolNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+    folder_id = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_name = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_path = fields.StringField(default=ROOT_FOLDER_NAME)
```

Comment:
Thanks, @felliott, for your review. I will be jumping right in today to address the issues you brought up.  One quick question right now.  You wrote:

> I think this needs some custom Fangorn config. When I click on my plan in the Files tab, I don't get any buttons in the fangorn toolbar. For readonly files, I'd expect to see "Download", "View", and "View on DMPTool.org" buttons. For the root folder, I'd expect a Download-As-Zip button.

Can you give me some tips on how to customize fangorn to get the actions you suggest? For example, maybe you can point to a good example in some other add-on with something like the features you describe?  Thanks!